### DISCUSSION TITLE
Stuck on lanaguage selection (no cloud-init usage)

### DISCUSSION THREAD

**Discussion by 3sky** [vmware/packer-examples-for-vsphere#784] [Help] [answered]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Project Version

latest

### VMware vSphere

8.0.2.0

### HashiCorp Packer

1.9.4

### HashiCorp Packer Plugin for VMware vSphere

1.2.2

### Guest Operating System

Ubuntu server 20.04 LTS

### Environment Details

New testing environment, 3 hosts, for testing packer build. 

### Description

I'm trying to build Ubuntu image, based on ISO from internet(`iso_url` added), and without serving server, so `common_data_source` = `disk`.
I stuck on loading cloud-init, it just stops on language selection window.

### Expected Behavior

Use cloud-init script(user-data) to auto install the system. 

### Actual Behavior

VM stuck on language selection window, however I'm able to see that under /dev/sr1 i have use-data as well as meta-data copied. 
Packer stuck on waiting for IP stage. 

### Steps to Reproduce

Use `disk` data_source, and iso from URL. No other changes from my side.

**Comment by tenthirtyam** [comment, marked as answer]
This issue would be best served on the plugin's repository as this is outside the scope of the examples based on the change to `iso_url`, which was removed from this project due to prior issues.

**Reply by tenthirtyam** [reply]
Moved here to the discussions as this is an ask for help, not a bug.

**Reply by 3sky** [reply]
Thanks @tenthirtyam, I was able to try build ubuntu without `iso_urls`, as it's not a hard requirement. And you're right, using "datastores" solved the issue. 
Thanks for that!

**Comment by JingChen23** [comment]
Hi @tenthirtyam , I ran into the same issue. I have never set `iso_urls` and my `common_data_source` is disk.

I have uploaded the ubuntu 20.04 image in the datastore, and use the path to load the iso.

What information should I provider firthur to let you know how did my issue happen?

**Reply by 3sky** [reply]
Hi, if you're using datastores, make sure that you provided correct `iso_path` variable. Default one it `iso_path = "iso/linux/ubuntu"`, and it's not strict documented, as thing to change.

**Reply by JingChen23** [reply]
Hi @3sky , I'm sure about that my iso_path is correct, because I add a "0" at the end of the iso file name then it leads to a different result.

![Screenshot 2023-11-29 at 20 51 13](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/15973672/d59ceca7-6673-4804-a86f-d866780c00e5)

This time it stucks at this point, not at the selecting language step.

**Reply by JingChen23** [reply]
It turns out that because I'm new to cloud init, I made a mistake when editing the cloud init file, I remove the first line comment.

```
#cloud-config
```

That leads to the cloud-init program cannot recogonize that the file as a cloud-config file.

**Reply by tenthirtyam** [reply]
That would do it! 😀
